logfilter/logfilter.go

```diff
@@ -162,6 +164,19 @@ func containsBlacklistedString(message string, blacklistedStrings []string) bool
 	return false
 }
 
+var apiKeyQueryParamRegExp = regexp.MustCompile("apiKey=[a-zA-Z0-9]+")
+
+const charactersToKeepVisible = 17
+
+func hideAPIKeysInURLQueryParams(msg string) string {
+	queryParams := apiKeyQueryParamRegExp.FindAllString(msg, -1)
+	for _, queryParam := range queryParams {
+		obscuredQueryParam := queryParam[:charactersToKeepVisible] + strings.Repeat("*", len(queryParam)-charactersToKeepVisible)
```

Comment:
Maybe we don't want to log the exact number of characters redacted, but just print a fixed number of asterisks (say, 8) instead.